Src/DirectXHelpers.cpp

```diff
@@ -93,3 +93,170 @@ void DirectX::CreateShaderResourceView(
 
     device->CreateShaderResourceView(tex, &srvDesc, srvDescriptor);
 }
+
+_Use_decl_annotations_
+void DirectX::CreateUnorderedAccessView(
+    ID3D12Device* device,
+    ID3D12Resource* tex,
+    D3D12_CPU_DESCRIPTOR_HANDLE uavDescriptor,
+    UINT mipLevel)
+{
+    const auto desc = tex->GetDesc();
+
+    D3D12_UNORDERED_ACCESS_VIEW_DESC uavDesc = {};
+    uavDesc.Format = desc.Format;
+
```

Comment:
In order to create the UAV descriptor, the original texture must have ``D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS``. Since we are providing an 'ease-of-use' wrapper here, you should add a check for each of these helpers to validate the scenario:


```
if ((desc.Flags & D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS) == 0)
{
    DebugTrace("ERROR: CreateUnorderedResourceView called on a resource created without support for UAVs");
   throw std::runtime_error("Requires D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS");
}
```

Similar checks in other functions for ``D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET`` and !``D3D12_RESOURCE_FLAG_DENY_SHADER_RESOURCE ``.